src/routes/safe/components/Apps/components/SafeAppsErrorBoundary.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, ErrorInfo } from 'react'
+import styled from 'styled-components'
+import { Text, Link, Icon, FixedIcon, Title } from '@gnosis.pm/safe-react-components'
+import { RouteComponentProps, withRouter } from 'react-router-dom'
+import { extractPrefixedSafeAddress, generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+
+type SafeAppsErrorBoundaryProps = {
+  children?: ReactNode
+} & RouteComponentProps
+
+type SafeAppsErrorBoundaryState = {
+  hasError: boolean
+  error?: Error
+}
+
+class SafeAppsErrorBoundary extends React.Component<SafeAppsErrorBoundaryProps, SafeAppsErrorBoundaryState> {
```

Comment:
Why have you used a class component?

Reply:
AFAIK you need to use class comps for ErrorBoundaries

From [react docs](https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries)

![image](https://user-images.githubusercontent.com/1576776/172328781-b0976f57-fd1d-45df-a92c-64a77cfce89e.png)